packages/clarity-js/src/data/metadata.ts

```diff
@@ -243,16 +243,54 @@ function getCookie(key: string): string {
       for (let i = 0; i < cookies.length; i++) {
         let pair: string[] = cookies[i].split(Constant.Equals);
         if (pair.length > 1 && pair[0] && pair[0].trim() === key) {
-          return pair[1];
+          // Some browsers automatically url encode cookie values if they are not url encoded.
+          // We therefore encode and decode cookie values ourselves.
+          return decodeCookieValue(pair[1]);
         }
       }
     }
   }
   return null;
 }
 
+function decodeCookieValue(value: string): string {
+  // For backwards compatability we need to consider 3 cases:
+  // * Cookie was previously not encoded by Clarity and browser did not encode it
+  // * Cookie was previously not encoded by Clarity and browser encoded it once or more
+  // * Cookie was previously encoded by Clarity and browser did not encode it
+  let [isEncoded, decodedValue] = valueIsEncoded(value);
+  if (!isEncoded) {
+    return value;
+  }
+
+  while (isEncoded) {
+    [isEncoded, decodedValue] = valueIsEncoded(decodedValue);
+  }
+
+  return decodedValue;
+}
+
+function valueIsEncoded(value: string): [boolean, string] {
```

Comment:
>valueIsEncoded [](http://example.com/codeflow?start=9&length=14)

nit: this name feels a little deceptive as it actually does the work to decode the cookie. Do you think it would be more clear if we inlined what you currently have as decodeCookieValue into the getCookieFunction (it's pretty small) and renamed valueIsEncoded to be decodeCookieValue?